src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -48,8 +48,14 @@ public static Command parse(String fullCommand) throws DukeException {
 
         switch (rootCommand.toLowerCase()) {
         case AddCommand.COMMAND_WORD:
-            checkAllowedArguments(argumentsMap, AddCommand.ALLOWED_ARGUMENTS);
-            return CommandCreator.createAddCommand(description, argumentsMap);
+            String subRootAddCommand = commandString.split(" ")[0];
+            if (subRootAddCommand.equals("link")) {
+                commandString = commandString.replaceFirst(subRootAddCommand, "").trim();
+                return CommandCreator.createAddCommand(commandString);
+            } else {
+                checkAllowedArguments(argumentsMap, AddCommand.ALLOWED_ARGUMENTS);
+                return CommandCreator.createAddCommand(description, argumentsMap);
+            }
```

Comment:
Maybe it will be better to move this to a new method called CommandCreator.parseAddCommand that returns a Command. Then put the rest of these code inside the method that will call either of the two `createAddCommand`s.